SpatialGDK/Source/SpatialGDKServices/Private/SpatialPackageManager.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialPackageManager.h"
+#include "SpatialGDKServicesConstants.h"
+
+
+DEFINE_LOG_CATEGORY(LogSpatialPackageManager);
+
+#define LOCTEXT_NAMESPACE "FSpatialPackageManager"
+
+FSpatialPackageManager::FSpatialPackageManager() {}
+
+void FSpatialPackageManager::Init() {}
+
+bool FSpatialPackageManager::TryFetchRuntimeBinary(FString RuntimeVersion)
+{
+
+	UE_LOG(LogSpatialPackageManager, Log, TEXT("RUNTIME VERSION %s"), *RuntimeVersion);
+
+	FString RuntimePath = FString::Printf(TEXT("%s/runtime/runtime.exe"), *SpatialGDKServicesConstants::GDKProgramPath);
+
+	// Check if the binary already exists for a given version
+	bool bSuccess = FPaths::FileExists(RuntimePath);
+
+	if (bSuccess)
+	{
+		UE_LOG(LogSpatialPackageManager, Log, TEXT("RUNTIME BINARIES ALREADY EXIST"));
+	}
+
+	// If it does not exist then fetch the binary using `spatial worker package retrieve`
+	// Download the zip to // UnrealGDK\SpatialGDK\Binaries\ThirdParty\Improbable\Programs\Runtime\*version* and unzip
+	else
+	{
+		FString RuntimeRetrieveArgs = FString::Printf(TEXT("package retrieve runtime x86_64-win32 %s runtime --unzip"), *RuntimeVersion);
+		FString RuntimeRetrieveResult;
+		int32 ExitCode;
+		FSpatialGDKServicesModule::ExecuteAndReadOutput(SpatialGDKServicesConstants::SpatialExe, RuntimeRetrieveArgs,
+														SpatialGDKServicesConstants::GDKProgramPath, RuntimeRetrieveResult, ExitCode);
```

Comment:
Let's augment the logic here so we can cache multiple versions of the runtime. Put each runtime version into it's own folder. This also means you need to check for the runtime versions folder in advance (for caching) and create it if it does not exist.